### PR TITLE
Fix typed property must not be accessed before initialization error when using --list option

### PR DESCRIPTION
**Pull request by iNem0o**
Hello

When using the `db-tools:restore --list` command, in the `listBackup()` method the `$this->restorer` variable is not inited yet which lead to a fatal error. 
https://github.com/makinacorpus/DbToolsBundle/blob/main/src/Command/RestoreCommand.php#L97

As the restorer is inited lately in the main process 
https://github.com/makinacorpus/DbToolsBundle/blob/main/src/Command/RestoreCommand.php#L110

i added the same call to the factory inside the `listBackups()` method to fix and tests case for the --list option

### CODE REVIEW

File: tests/Functional/Command/RestoreCommandTest.php [hunk 1]
```diff
@@ -7,6 +7,11 @@
 use MakinaCorpus\DbToolsBundle\Test\FunctionalKernelTestCase;
 use Symfony\Bundle\FrameworkBundle\Console\Application;
 use Symfony\Component\Console\Tester\CommandTester;
+use MakinaCorpus\DbToolsBundle\Command\RestoreCommand;
+use MakinaCorpus\DbToolsBundle\Restorer\AbstractRestorer;
+use MakinaCorpus\DbToolsBundle\Restorer\RestorerFactory;
+use MakinaCorpus\DbToolsBundle\Storage\Storage;
+use PHPUnit\Framework\TestCase;
```
Comment (pounard) [inline tests/Functional/Command/RestoreCommandTest.php:14]: Please do an alphabetical sort of imports.

File: src/Bridge/Symfony/DependencyInjection/Compiler/DbToolsPass.php [hunk 2]
```diff
@@ -29,7 +29,7 @@ public function process(ContainerBuilder $container): void
             }
 
             if (isset($config['anonymization']) && isset($config['anonymization']['yaml'])) {
-                foreach($config['anonymization']['yaml'] as $connectionName => $file) {
+                foreach ($config['anonymization']['yaml'] as $connectionName => $file) {
```
Comment (pounard) [inline src/Bridge/Symfony/DependencyInjection/Compiler/DbToolsPass.php:32]: I'm OK with all those changes, we'll merge with it, but please avoid polluting diffs with code style in the future. We do run cs fixer from time to time and make it explicit in commits.

File: tests/Functional/Command/RestoreCommandTest.php [hunk 3]
```diff
@@ -26,4 +31,103 @@ public function testExecute(): void
 
         self::assertCommandIsSuccessful($commandTester);
     }
+
+    /**
+     * Creates a command tester with mocked dependencies for listBackups method
+     *
+     * @param array $backupFiles List of backup files to be returned by storage
```
Comment (pounard) [inline tests/Functional/Command/RestoreCommandTest.php:38]: Nitpick: we put the human readable documentation in the next line.

**Comment by pounard** [comment]
Hello @iNem0o, your PR seems fine, please fix the minor comments and it's all good.

Pipeline failure seems unrelated, we'll see how it behave on your next push. If it continues to fail I'll pull your branch and test on my side to be sure it's completely unrelated.

File: src/Bridge/Symfony/DependencyInjection/Compiler/DbToolsPass.php (hunk 2 above)
Comment (SimonMellerin) [reply src/Bridge/Symfony/DependencyInjection/Compiler/DbToolsPass.php:32]: Here, it's probably a bit our fault: the [contribution rules](https://dbtoolsbundle.readthedocs.io/en/stable/contribute/guide.html) explicitly ask you to run `./dev.sh checks` before committing.

What's weird is that the CI shouldn't have accepted to merge these modifications that don't respect the code style. There must be a bug in the CI :shrug:

**Comment by iNem0o** [comment]
Hello @pounard, I have made the requested changes as per your feedback.
Let me know if there are any further adjustments needed after the CI checks.

**Comment by pounard** [comment]
@iNem0o thanks, I'll review this tomorrow.

**Comment by pounard** [comment]
@iNem0o Thanks! I'll add a line in the changelog file.

**Pull request merged**